lib/api/api-request.js

```diff
@@ -30,6 +57,91 @@ class RequestResponse extends Readable {
   }
 }
 
+class Body {
+  constructor (resume, abort) {
+    this[kAbort] = abort
+    this[kResume] = resume
+    this[kBody] = new RequestDuplex(this[kResume], this[kAbort]).on('error', () => {})
+    this[kUsed] = false
+  }
+
+  [kPush] (chunk) {
+    return this[kBody].push(chunk)
+  }
+
+  [kDestroy] (err) {
+    this[kBody].destroy(err)
+  }
+
+  get stream () {
+    if (this[kBody].readableDidRead) {
+      throw new TypeError('disturbed')
+    }
+
+    return this[kBody]
+  }
```

Comment:
This is almost like `BodyMixin` from the web standard with the small exception of how to consume as Node stream.... anyone got suggestions on how that could/should look? @szmarczak @Ethan-Arrowood @mcollina @dnlup